exercises/concept/savings-account/savings_account_test.rb

```diff
@@ -102,4 +103,12 @@ def test_years_before_desired_balance_for_large_start_balance
   def test_years_before_desired_balance_for_large_difference_between_start_and_desired_balance
     assert_equal 85, SavingsAccount.years_before_desired_balance(2_345.67, 12_345.678_9)
   end
+
+  def test_years_before_desired_balance_for_negative_current_balance
+    assert_raises(ArgumentError, 'Current balance cannot be negative') do
+      Timeout.timeout(0.1) do
+        SavingsAccount.years_before_desired_balance(-1.0, 1.0)
```

Comment:
This also has the problem of a balance that is 0.  More than "can not be negative", it should probably state that it should be positive.

For communicating the problem. we might also rescue the timeout and raise `RuntimeError, 'Impossible Solution'`

It might look like this:

```ruby
  def test_years_before_desired_balance_for_negative_current_balance
    assert_raises(ArgumentError, 'Current balance cannot be negative') do
      Timeout.timeout(0.1) do
        SavingsAccount.years_before_desired_balance(0, 1.0)
      end
    rescue Timeout::Error
      raise RuntimeError, 'Impossible Solution, raise an ArgumentError if current balance is not positive.'
    end
  end
```

Reply:
I think I prefer the "must be positive" rather than "cannot be negative", as that includes 0.